### PR TITLE
[NOD-1316] Refactor TestGHOSTDAG to enable arbitrary DAGs

### PR DESCRIPTION
**Pull request by elichai**
This will allow me and other researchers to easily create new DAGs (using a tool I wrote in automation) and add them to the test suite with a simple copy-paste

`VirtualBlues()` was added and is a blocker for the DAG creation tool:
https://github.com/kaspanet/automation/blob/c81e5dd2fe0e0ca6c95e81742592bf982b575911/stuff/dagtests/dagtests.go#L239

### CODE REVIEW

File: domain/blockdag/dag.go [hunk 1]
```diff
@@ -281,6 +281,15 @@ func (dag *BlockDAG) SelectedTipBlueScore() uint64 {
 	return dag.selectedTip().blueScore
 }
 
+// VirtualBlues returns the blues of the current virtual block
+func (dag *BlockDAG) VirtualBlues() []*daghash.Hash {
```
Comment (stasatdaglabs) [inline domain/blockdag/dag.go:285]: Please rename this to VirtualBlueHashes.

File: domain/blockdag/ghostdag_test.go [hunk 2]
```diff
@@ -188,41 +66,42 @@ func TestGHOSTDAG(t *testing.T) {
...
-				for _, parentID := range blockData.parents {
+				for _, parentID := range blockData.Parents {
 					parent := blockByIDMap[parentID]
 					parents.add(parent)
 				}
 
 				block, err := PrepareBlockForTest(dag, parents.hashes(), nil)
 				if err != nil {
-					t.Fatalf("TestGHOSTDAG: block %v got unexpected error from PrepareBlockForTest: %v", blockData.id, err)
+					t.Fatalf("TestGHOSTDAG: block %v got unexpected error from PrepareBlockForTest: %v", blockData.ID,
```
Comment (stasatdaglabs) [inline domain/blockdag/ghostdag_test.go:81]: Please use %s for strings rather than %v, here and everywhere.

**Review by stasatdaglabs — CHANGES_REQUESTED** [review]

File: domain/blockdag/dag.go [hunk 3]
```diff
@@ -281,6 +281,15 @@ func (dag *BlockDAG) SelectedTipBlueScore() uint64 {
...
+	hashes := make([]*daghash.Hash, len(dag.virtual.blues))
```
[the remark this answers is not in the document]
Comment (elichai) [reply domain/blockdag/dag.go:288]: Done

File: domain/blockdag/ghostdag_test.go (hunk 2 above)
Comment (elichai) [reply domain/blockdag/ghostdag_test.go]: Done

File: domain/blockdag/dag.go (hunk 1 above)
Comment (elichai) [reply domain/blockdag/dag.go]: Done

**Review by someone235 — APPROVED** [review]

**Review by stasatdaglabs — APPROVED** [review]

**Comment by elichai** [comment]
That last commit should be reviewed cautiously because unlike the rest, this one touches consensus code.
it basically stores the flags passed when adding a delayedBlock s.t. it can use those flags when the delay expires
https://github.com/kaspanet/kaspad/pull/899/commits/690244a98324c28bd73e435d70354b08b39370e6

**Review by stasatdaglabs — APPROVED** [review]

**Pull request merged**